### PR TITLE
feat: Add non blocking way of getting values

### PR DESCRIPTION
**Pull request by vahidlazio**

### CODE REVIEW

File: android/src/main/kotlin/com/example/confidence_flutter_sdk/ConfidenceFlutterSdkPlugin.kt [hunk 1]
```diff
@@ -95,10 +100,18 @@ class ConfidenceFlutterSdkPlugin: FlutterPlugin, MethodCallHandler, ActivityAwar
         val value = confidence.getValue(key, defaultValue)
         result.success(Json.encodeToString(NetworkConfidenceValueSerializer, value))
       }
+      "readAllFlags" -> {
+        val flags = readAllFlags()
+        val map = mutableMapOf<String, ConfidenceValue>()
+        flags.flags.associateByTo(map, { it.flag }, { ConfidenceValue.Struct(it.value) })
+        print(Json.encodeToString(NetworkConfidenceValueSerializer, ConfidenceValue.Struct(map)))
```
Comment (nicklasl) [inline android/src/main/kotlin/com/example/confidence_flutter_sdk/ConfidenceFlutterSdkPlugin.kt:107]: remove print

**Review by nicklasl — COMMENTED** [review]
Looks good but I think we need to give this one more round where we also make sure the dart cache refreshes after a "reconciling event" -- when new flags have been fetched due to a `putContext`.

**Comment by vahidlazio** [comment]
@nicklasl for putContext, we do fetch and activate [here](https://github.com/spotify/confidence-sdk-flutter/pull/24/files#diff-9b270f285c0366ea7184dde91a592377df2afde17302d298382be139fc5205b3R14).

**Comment by nicklasl** [comment]
> @nicklasl for putContext, we do fetch and activate [here](https://github.com/spotify/confidence-sdk-flutter/pull/24/files#diff-9b270f285c0366ea7184dde91a592377df2afde17302d298382be139fc5205b3R14).

hmm.. so double fetching? (or is the putContext not triggering a fetch on the "native" side?)

**Comment by vahidlazio** [comment]
@nicklasl yes double fetching for now.

**Review by nicklasl — APPROVED** [review]

**Pull request merged**